backend/routes/searchApiFactory.test.js

```diff
@@ -0,0 +1,156 @@
+
+      await searchApi({
+        locals: { user: { maintainAccessAdmin: true } },
+        user: 'jane',
+        groupCode: 'MDI',
+        roleCode: 'OMIC_ADMIN',
+        activeCaseload: 'LEI',
+        status: 'ACTIVE',
+        pageSize: 20,
+        pageOffset: 40,
+      })
+
+      expect(prisonApi.userSearchAdmin).toBeCalledWith(
+        {
+          user: { maintainAccessAdmin: true },
+          requestHeaders: {
+            'page-limit': 20,
+            'page-offset': 40,
+          },
+        },
+        {
+          nameFilter: 'jane',
+          roleFilter: 'OMIC_ADMIN',
+          status: 'ACTIVE',
+          caseload: 'MDI',
+          activeCaseload: 'LEI',
+        },
+      )
+    })
+    it('will call normal search without admin role context and not pass caseload filter', async () => {
+      prisonApi.userSearch.mockResolvedValue([])
+
+      await searchApi({
+        locals: { user: { maintainAccessAdmin: false } },
+        user: 'jane',
+        groupCode: 'MDI',
+        roleCode: 'OMIC_ADMIN',
+        activeCaseload: 'LEI',
+        status: 'ACTIVE',
+        pageSize: 20,
+        pageOffset: 40,
+      })
+
+      expect(prisonApi.userSearch).toBeCalledWith(
+        {
+          user: { maintainAccessAdmin: false },
+          requestHeaders: {
+            'page-limit': 20,
+            'page-offset': 40,
+          },
+        },
+        {
+          nameFilter: 'jane',
+          roleFilter: 'OMIC_ADMIN',
+          status: 'ACTIVE',
+        },
+      )
+    })
+    it('will retrieve emails for each user found and add to results', async () => {
+      prisonApi.userSearchAdmin.mockResolvedValue([
+        { username: 'jane', activeCaseLoadId: 'MDI' },
+        { username: 'jill', activeCaseLoadId: 'MDI' },
+      ])
+      oauthApi.userEmails.mockResolvedValue([{ username: 'jane', email: 'jane@email.com' }])
+
+      const results = await searchApi({
+        locals: {
+          user: { maintainAccessAdmin: true },
+        },
+        pageSize: 20,
+        pageOffset: 40,
+      })
+
+      expect(oauthApi.userEmails).toBeCalledWith(
+        {
+          user: { maintainAccessAdmin: true },
+          requestHeaders: {
+            'page-limit': 20,
+            'page-offset': 40,
+          },
+        },
+        ['jane', 'jill'],
+      )
+      expect(results).toEqual([
+        { username: 'jane', activeCaseLoadId: 'MDI', email: 'jane@email.com' },
+        { username: 'jill', activeCaseLoadId: 'MDI' },
+      ])
+    })
+  })
+  describe('searchableRoles', () => {
+    it('will get admin roles with admin role context', async () => {
+      prisonApi.getRolesAdmin.mockResolvedValue([])
+
+      await searchableRoles({ user: { maintainAccessAdmin: true } })
+
+      expect(prisonApi.getRolesAdmin).toBeCalledWith({
+        user: { maintainAccessAdmin: true },
+      })
+    })
+    it('will get normal roles with admin role context', async () => {
```

Comment:
Small typo - should be `without admin role context`?

Reply:
thanks